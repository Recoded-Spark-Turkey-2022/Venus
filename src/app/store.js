// eslint-disable-next-line no-unused-vars
import { configureStore, getDefaultMiddleware } from '@reduxjs/toolkit';
import listingReducer from '../features/listings/listingSlice';

export const store = configureStore({
  reducer: {
    listing: listingReducer,
  },
  // eslint-disable-next-line no-shadow
  middleware: (getDefaultMiddleware) =>
    getDefaultMiddleware({
      serializableCheck: false,
    }),
});
