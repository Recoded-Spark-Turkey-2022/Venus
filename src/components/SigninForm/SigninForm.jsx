import React from "react";
import { useForm } from "react-hook-form";



const SignInForm= ()=> {

    const { register, handleSubmit } = useForm();

    const onSubmit = (data)=> {
        console.log(data);
    }

    return(
       
         <form className="flex flex-col gap-4 items-center justify-center py-1 md:py-3  " onSubmit={handleSubmit(onSubmit)}>
           
           <input 
           placeholder="Your Email"
           {...register("email", { required: true })} 
           className="border border-gray-300 px-0 md:px-3 py-1 md:py-2 text-gray-900 placeholder-gray-500 rounded w-32 md:w-64"/>
           <input type="password"
           placeholder="Password" 
           {...register("password", { required: true, minLength: 6 })} 
           className="border border-gray-300 px-0 md:px-3 py-1 md:py-2 text-gray-900 placeholder-gray-500 rounded w-32 md:w-64"/>
           <button type="submit" id="mediumBlue-button" className="px-8 whitespace-nowrap rounded-2xl py-0 md:py-1 w-16 md:w-32 flex justify-center items-center">
            Sign In</button>

        </form>

    )
}
export default SignInForm;