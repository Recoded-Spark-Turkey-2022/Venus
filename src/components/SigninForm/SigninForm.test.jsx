import React from 'react';
import renderer from "react-test-renderer";
import { Provider } from 'react-redux';
import { store } from '../../app/store';
import { BrowserRouter as Router } from "react-router-dom";
import SigninForm from "./SigninForm";

jest.mock('react-i18next', () => ({
    useTranslation: () => {
      return {
        t: (str) => str
      };
    },
  }));

  it(`renders correctly` , () => {
    const tree = renderer.create(
    <Router>
      <Provider store={store}>
        <SigninForm />
        </Provider>
    </Router>);
    expect(tree).toMatchSnapshot();
});
