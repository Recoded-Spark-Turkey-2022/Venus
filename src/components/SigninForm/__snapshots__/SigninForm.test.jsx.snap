// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<form
  className="flex flex-col gap-4 items-center justify-center pt-2 pb-10 md:py-3   rounded-2xl"
  onSubmit={[Function]}
>
  <input
    aria-invalid="false"
    className="border border-gray-300  w-full md:w-[80%]  px-0 md:px-3 m-0 py-1 md:py-2 text-gray-900 placeholder-gray-500 rounded"
    name="email"
    onBlur={[Function]}
    onChange={[Function]}
    placeholder="Your Email"
  />
  <input
    aria-invalid="false"
    className="border border-gray-300 px-0 md:px-3 py-1 md:py-2 text-gray-900 m-0 placeholder-gray-500 rounded w-full md:w-[80%]"
    name="password"
    onBlur={[Function]}
    onChange={[Function]}
    placeholder="Password"
    type="password"
  />
  <button
    className="px-8 whitespace-nowrap w-[65%] l md:w-44 rounded-2xl py-2 md:py-1 flex justify-center items-center"
    id="mediumBlue-button"
    type="submit"
  >
    Button.si
  </button>
</form>
`;
